src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -297,7 +330,11 @@ object ScalafixPlugin extends AutoPlugin {
     val parser = Def.setting(
       new ScalafixCompletions(
         workingDirectory = (ThisBuild / baseDirectory).value.toPath,
-        loadedRules = () => scalafixInterfaceProvider.value().availableRules(),
+        loadedRules = () =>
+          scalafixInterfaceProvider
+            // sbt Credentials is a task, so unfortunately it cannot be showed up here
+            .value((ThisBuild / scalafixResolvers).value)
+            .availableRules(),
```

Comment:
I'm not sure if I get your question correctly, AFAIK, It should work exactly like before since the only things I do is make an extra argument for `scalafixInterfaceProvider`, and pass what originally inside it to the function.

Though I'm not very familiar with sbt plugin, am I miss something here?

Reply:
Or we can recover by reconstruct a `Scalafix` instance with no external jars, but this makes it's inappropriate to recover in `loadRules` since it breaks cache.

It could be done inside ` ScalafixInterface.fromToolClasspath`, but that will change the behavior for all tasks not just `ScalafixCompletion`, otherwise we probably have to pass a flag into it to indicate it's used by constructing `ScalafixCompletion`.

Reply:
Implemented, I used `scala.util.Try` instead of try syntax since it looks more compact.